internal/controlplane/handlers_githubwebhooks.go

```diff
@@ -240,33 +267,124 @@ func (s *Server) HandleGitHubWebHook() http.HandlerFunc {
+		// The following events are not available in go-github
+		// and must be handled manually.
+		case "branch_protection_configuration":
+			res, processingErr = s.processBranchProtectionConfigurationEvent(ctx, rawWBPayload)
+		case "repository_advisory":
+			res, processingErr = s.processRepositoryAdvisoryEvent(ctx, rawWBPayload)
+		case "repository_ruleset":
+			res, processingErr = s.processRepositoryRulesetEvent(ctx, rawWBPayload)
+		case "secret_scanning_alert_location":
+			res, processingErr = s.processSecretScanningAlertLocationEvent(ctx, rawWBPayload)
+		case "package":
+			// This is an artifact-related event, and can
+			// only trigger a reconciliation.
+			res, processingErr = s.processPackageEvent(ctx, rawWBPayload)
+		}
+
+		switch event := event.(type) {
+		case *github.PingEvent:
+			// For ping events, we do not set wes.Accepted
+			// to true because they're not relevant
+			// business events.
+			wes.Accepted = false
+			wes.Error = false
+			s.processPingEvent(ctx, event)
+		case *github.MetaEvent:
+			// As per github documentation, MetaEvent is
+			// triggered when the webhook that this event
+			// is configured on is deleted.
+			//
+			// Our action here is to de-register the
+			// related repo.
+			res, processingErr = s.processMetaEvent(ctx, event)
+
+		// All these events are related to a repo and usually
+		// contain an action. They all trigger a
+		// reconciliation or, in some cases, a deletion.
+		case *github.BranchProtectionRuleEvent:
+			res, processingErr = s.processBranchProtectionRuleEvent(ctx, event)
+		case *github.CodeScanningAlertEvent:
+			res, processingErr = s.processCodeScanningAlertEvent(ctx, event)
+		case *github.CreateEvent:
+			res, processingErr = s.processCreateEvent(ctx, event)
+		case *github.MemberEvent:
+			res, processingErr = s.processMemberEvent(ctx, event)
+		case *github.PublicEvent:
+			res, processingErr = s.processPublicEvent(ctx, event)
+		case *github.RepositoryEvent:
+			res, processingErr = s.processRepositoryEvent(ctx, event)
+		case *github.RepositoryImportEvent:
+			res, processingErr = s.processRepositoryImportEvent(ctx, event)
+		case *github.SecretScanningAlertEvent:
+			res, processingErr = s.processSecretScanningAlertEvent(ctx, event)
+		case *github.TeamAddEvent:
+			res, processingErr = s.processTeamAddEvent(ctx, event)
+		case *github.TeamEvent:
+			res, processingErr = s.processTeamEvent(ctx, event)
+		case *github.RepositoryVulnerabilityAlertEvent:
+			res, processingErr = s.processRepositoryVulnerabilityAlertEvent(ctx, event)
+		case *github.SecurityAdvisoryEvent:
+			res, processingErr = s.processSecurityAdvisoryEvent(ctx, event)
+		case *github.SecurityAndAnalysisEvent:
+			res, processingErr = s.processSecurityAndAnalysisEvent(ctx, event)
+		case *github.OrgBlockEvent,
+			*github.OrganizationEvent:
+			l.Info().Msgf("webhook events %s do not contain repo", wes.Typ)
+		case *github.PushEvent:
+			// TODO has GetRepo but the type is different :|
```

Comment:
This needs to be handled so we can keep evaluating `git` ingester policies.

Reply:
Resolved, I tested this here https://github.com/blkt/demo-repo-python/pull/2

Reply:
Also
* [code](https://github.com/stacklok/minder/blob/a26fafdee87e9bbf06322fe9fad1cc61d7476cc9/internal/controlplane/handlers_githubwebhooks.go#L475)
* [tests](https://github.com/stacklok/minder/blob/a26fafdee87e9bbf06322fe9fad1cc61d7476cc9/internal/controlplane/handlers_githubwebhooks_test.go#L2573-L2642)